pkg/trace/agent/agent.go

```diff
@@ -83,7 +89,8 @@ func NewAgent(ctx context.Context, conf *config.AgentConfig) *Agent {
 		EventProcessor:        newEventProcessor(conf),
 		TraceWriter:           writer.NewTraceWriter(conf),
 		StatsWriter:           writer.NewStatsWriter(conf, statsChan),
-		obfuscator:            obfuscate.NewObfuscator(conf.Obfuscation),
+		obfuscator:            obfuscate.NewObfuscator(oconf),
+		cardObfuscator:        newCreditCardsObfuscator(conf.Obfuscation.CreditCards),
```

Comment:
why isn't the cardObfuscator part of the obfuscator?

Reply:
Because it needs to add a hook into the `pb` packages which obfuscates certain tags at decode time. It is very tightly couple with the payload decoder and with the Span. The actual function which checks if a string is a credit card number is part of `pkg/obfuscate` under `IsCardNumber`.